tests/rados/test_reads_balancer.py

```diff
@@ -0,0 +1,273 @@
+                )
+
+                # todo: uncomment the below exception once the bug is fixed
+                # raise Exception("Scores out of bound failure")
+
+            log.debug("Checking if the scores are calculated properly")
+            # Balance_score = raw_score / Optimal_score
+            expected_score = (
+                read_scores["raw_score_acting"] / read_scores["optimal_score"]
+            )
+            if round(read_scores["score_acting"], 2) != round(expected_score, 2):
+                log.error(
+                    f"The scores for the pool is not calculated properly. acting score {read_scores['score_acting']}"
+                    f"Expected : {(read_scores['raw_score_acting'] / read_scores['optimal_score'])}"
+                )
+                raise Exception("Scores out of bound failure")
+            log.info("Completed checking the balance scores for the pool. No errors")
+        else:
+            log.info(
+                f"Selected pool {pool_name} is a EC pool, skipping check of balance scores."
+            )
+        log.info("Completed verification of balance scores for all the pools")
+
+    log.debug(
+        "Setting config to allow clients to perform read balancing on the clusters."
+    )
+    config_cmd = "ceph osd set-require-min-compat-client reef"
+    rados_obj.run_ceph_command(cmd=config_cmd)
+    time.sleep(5)
+
+    if config.get("online_command_verification"):
+        log.info(
+            "Verifying the usage of online commands to change the acting set of the pool"
+        )
+        for pool_name in existing_pools:
+            pool_details = rados_obj.get_pool_details(pool=pool_name)
+            log.debug(f"Selected pool name : {pool_name}")
+
+            # Testing on replicated pools
+            if not pool_details["erasure_code_profile"]:
+                pool = pool_details["pool_name"]
+
+                # Fetching the acting set for the pool
+                # Getting a sample PG from the pool
+                pool_id = pool_obj.get_pool_id(pool_name=pool)
+                pgid = f"{pool_id}.0"
+                acting_set = rados_obj.get_pg_acting_set(pg_num=pgid)
+
+                log.debug(f"Acting set for PG-id {pgid} in pool {pool} is {acting_set}")
+
+                # Trying to change the primary with existing primary. -ve test
+                new_osd = acting_set[0]
+                change_cmd = f"ceph osd pg-upmap-primary {pgid} {new_osd}"
+                error_str = (
+                    f"Error EINVAL: osd.{new_osd} is already primary for pg {pgid}"
+                )
+
+                try:
+                    out, err = client_node.exec_command(sudo=True, cmd=change_cmd)
+                except Exception as e:
+                    if error_str not in str(e).strip():
+                        log.error(
+                            "Correct error string not seen in stderr stream"
+                            f"Expected : {error_str} \n Got : {e}"
```

Comment:
f"Expected : {error_str} \n Got : {_**err**_}"

Reply:
Both are same. does not matter either way.